File: Examples/statistics/codes/prior.cpp
// ======================================
// Example code: how to construct a prior
// ======================================

#include "PriorDistribution.h"

int main () {

  try {
    
    double xmin = -10., xmax = 10.;
    double mean = -1., sigma = 0.1;
    
    cbl::statistics::PriorDistribution priorDistribution { cbl::glob::DistributionType::_Gaussian_, {mean, sigma}, xmin, xmax };
    
    for (int i=0; i<100; ++i) {
      double value = priorDistribution.sample();
      std::cout << i << "  " << value << "  " << priorDistribution(value) << std::endl;
    }

  }

  catch(cbl::glob::Exception &exc) { std::cerr << exc.what() << std::endl; exit(1); }
  
  return 0;
}

 

File: Examples/statistics/codes/prior.py
import numpy as np
import matplotlib.pyplot as plt
import CosmoBolognaLib as cbl

xmin = -10.
xmax = 10.
mean = -1.
sigma = 0.1

prior = cbl.PriorDistribution(cbl.DistributionType__Gaussian_, [mean, sigma], xmin, xmax)

nExtr = 2000
sample = np.array([prior.sample(i) for i in range(nExtr)])
psample = np.array([prior(ss) for ss in sample])

plt.hist(sample, 20, normed=True)
plt.plot(sample, psample, '.')
plt.show(block=False)
